### DISCUSSION TITLE
Repository dispatch event - query workflow status

### DISCUSSION THREAD

**Discussion by adrianmiron** [community/community#24918] [Actions]
<p>Hi,</p>
<p>I am triggering some repository_dispatch events from spinnaker and it all works fine.</p>
<p>I am however at a loss on how to query the workflow final status ( the webhook response returns an empty message with 204 response code,  immediately after accepting it, as the documentation states).</p>
<p>Can i somehow get the run_id from the webhook response so that i can query that particular run_id for it’s status ? Or is there some other way to do this ?</p>
<p>Thanks,<br>
Adrian</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/adrianmiron">@adrianmiron</a>, welcome to the GitHub Support Community! There isn’t an easy way to do this at the moment unfortunately. The reason the run ID isn’t returned on that API request it because this event doesn’t directly <em>trigger</em> a run, it just sends an event, and the information for the workflow that’s triggered doesn’t exist inside the API yet.</p>
<p>The best thing you can do is to poll the <a href="https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository">workflow runs API</a> and filter to the workflow event <code>workflow_dispatch</code>  and branch used to trigger the workflow dispatch.</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository" target="_blank" rel="noopener">Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository" target="_blank" rel="noopener">//docs.github.com/en/rest/reference/actions</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by adrianmiron** [comment]
<p>Thanks for the confirmation, i found sort of a workaround.</p>

**Reply by lvijnck** [reply]
Mind sharing what that is?

**Reply by iteaguy** [reply]
Hey @adrianmiron - I am interested in the same thing. Would you mind sharing what your workaround was? Greatly appreciated.

Also, if you are making a `repository_dispatch` event in Github, you are using a PAT, which I am sure is exposed in the execution context. Do you know of a way to hide/suppress that?

**Reply by adrianmiron** [reply]
Wow, it's been a 2 and 1/2 years and i no longer work with spinnaker for about 2 of them....but from what i remeber i listed the run_id's for that particular workflow file and grabbed the latest one ( this works if you set max concurrency to one per particular workflow; this means only one will run at a particular time while the rest will be queued and will run afterwards)

**Comment by sambshapiro** [comment]
Similar requests here:
https://github.com/orgs/community/discussions/27226
https://github.com/orgs/community/discussions/17389
